Build fails on kernel 6.15

I'll try fixing this
```
Building module(s)
# command: make -j16 KERNELRELEASE=6.15.0-rc1-00060-ga24588245776 -C /usr/lib/modules/6.15.0-rc1-00060-ga24588245776/build M=/var/lib/dkms/xone/0.3.r101.gaeb27e6/build
make: Entering directory '/usr/lib/modules/6.15.0-rc1-00060-ga24588245776/build'
make[1]: Entering directory '/var/lib/dkms/xone/0.3.r101.gaeb27e6/build'
  CC [M]  transport/wired.o
  CC [M]  transport/dongle.o
  CC [M]  transport/mt76.o
  CC [M]  bus/bus.o
  CC [M]  bus/protocol.o
  CC [M]  auth/auth.o
  CC [M]  auth/crypto.o
  CC [M]  driver/common.o
  CC [M]  driver/gamepad.o
  CC [M]  driver/headset.o
  CC [M]  driver/chatpad.o
  CC [M]  driver/madcatz_strat.o
  CC [M]  driver/madcatz_glam.o
  CC [M]  driver/pdp_jaguar.o
driver/gamepad.c: In function ‘gip_gamepad_init_input’:
driver/gamepad.c:210:9: error: implicit declaration of function ‘del_timer_sync’ [-Wimplicit-function-declaration]
  210 |         del_timer_sync(&gamepad->rumble.timer);
      |         ^~~~~~~~~~~~~~
make[3]: *** [/usr/lib/modules/6.15.0-rc1-00060-ga24588245776/build/scripts/Makefile.build:203: driver/gamepad.o] Error 1
make[3]: *** Waiting for unfinished jobs....
driver/headset.c: In function ‘gip_headset_probe’:
driver/headset.c:502:9: error: implicit declaration of function ‘hrtimer_init’; did you mean ‘hrtimers_init’? [-Wimplicit-function-declaration]
  502 |         hrtimer_init(&headset->timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
      |         ^~~~~~~~~~~~
      |         hrtimers_init
make[3]: *** [/usr/lib/modules/6.15.0-rc1-00060-ga24588245776/build/scripts/Makefile.build:203: driver/headset.o] Error 1
make[2]: *** [/usr/lib/modules/6.15.0-rc1-00060-ga24588245776/build/Makefile:2009: .] Error 2
make[1]: *** [/usr/lib/modules/6.15.0-rc1-00060-ga24588245776/build/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/var/lib/dkms/xone/0.3.r101.gaeb27e6/build'
make: *** [Makefile:248: __sub-make] Error 2
make: Leaving directory '/usr/lib/modules/6.15.0-rc1-00060-ga24588245776/build'

# exit code: 2
# elapsed time: 00:00:02
```

--- a/driver/gamepad.c
+++ b/driver/gamepad.c
@@ -6,6 +6,7 @@
 #include <linux/module.h>
 #include <linux/uuid.h>
 #include <linux/timer.h>
+#include <linux/version.h>
 
 #include "common.h"
 #include "../auth/auth.h"
@@ -207,8 +208,11 @@ static int gip_gamepad_init_input(struct gip_gamepad *gamepad)
 	return 0;
 
 err_delete_timer:
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,0)
+	timer_delete_sync(&gamepad->rumble.timer);
+#else
 	del_timer_sync(&gamepad->rumble.timer);
-
+#endif
 	return err;
 }
 
@@ -335,7 +339,11 @@ static void gip_gamepad_remove(struct gip_client *client)
 {
 	struct gip_gamepad *gamepad = dev_get_drvdata(&client->dev);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,0)
+	timer_delete_sync(&gamepad->rumble.timer);
+#else
 	del_timer_sync(&gamepad->rumble.timer);
+#endif
 }
 
 static struct gip_driver gip_gamepad_driver = {

--- a/driver/headset.c
+++ b/driver/headset.c
@@ -6,6 +6,7 @@
 #include <linux/module.h>
 #include <linux/hrtimer.h>
 #include <linux/vmalloc.h>
+#include <linux/version.h>
 #include <sound/core.h>
 #include <sound/initval.h>
 #include <sound/pcm.h>
@@ -499,8 +500,13 @@ static int gip_headset_probe(struct gip_client *client)
 	INIT_DELAYED_WORK(&headset->work_power_on, gip_headset_power_on);
 	INIT_WORK(&headset->work_register, gip_headset_register);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,0)
+	hrtimer_setup(&headset->timer, gip_headset_send_samples,
+		      CLOCK_MONOTONIC, HRTIMER_MODE_REL);
+#else
 	hrtimer_init(&headset->timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
 	headset->timer.function = gip_headset_send_samples;
+#endif
 
 	err = gip_enable_audio(client);
 	if (err)

